lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/context/ContextRepresenters.java

```diff
@@ -210,7 +227,7 @@ private synchronized void onNewContext(final ReefServiceProtos.ContextStatusProt
         Optional.of(contextStatusProto.getParentId()) : Optional.<String>empty();
     final EvaluatorContext context = contextFactory.newContext(contextID, parentID);
     this.addContext(context);
-    if (contextStatusProto.getRecovery()) {
+    if (DriverRestartUtilities.isRestartAndIsPreviousEvaluator(driverRestartManager, context.getEvaluatorId())) {
```

Comment:
I'd prefer to leave it as a static method, then it won't be up to the DriverRestartManager implementation to determine how to implement the method, since it's essentially just the concatenation of two method calls.